--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) Contributors to the OpenEXR Project.
+
+cc_library(
+    name = "deflate",
+    srcs = [
+        "common_defs.h",
+        "lib/adler32.c",
+        "lib/arm/adler32_impl.h",
+        "lib/arm/cpu_features.c",
+        "lib/arm/cpu_features.h",
+        "lib/arm/crc32_impl.h",
+        "lib/arm/crc32_pmull_helpers.h",
+        "lib/arm/crc32_pmull_wide.h",
+        "lib/arm/matchfinder_impl.h",
+        "lib/bt_matchfinder.h",
+        "lib/cpu_features_common.h",
+        "lib/crc32_multipliers.h",
+        "lib/crc32_tables.h",
+        "lib/decompress_template.h",
+        "lib/deflate_compress.c",
+        "lib/deflate_compress.h",
+        "lib/deflate_constants.h",
+        "lib/deflate_decompress.c",
+        "lib/gzip_constants.h",
+        "lib/hc_matchfinder.h",
+        "lib/ht_matchfinder.h",
+        "lib/lib_common.h",
+        "lib/matchfinder_common.h",
+        "lib/utils.c",
+        "lib/x86/adler32_impl.h",
+        "lib/x86/adler32_template.h",
+        "lib/x86/cpu_features.c",
+        "lib/x86/cpu_features.h",
+        "lib/x86/crc32_impl.h",
+        "lib/x86/crc32_pclmul_template.h",
+        "lib/x86/decompress_impl.h",
+        "lib/x86/matchfinder_impl.h",
+        "lib/zlib_compress.c",
+        "lib/zlib_constants.h",
+        "lib/zlib_decompress.c",
+    ],
+    hdrs = ["libdeflate.h"],
+    includes = ["."],
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "libdeflate",
+    actual = ":deflate",
+    visibility = ["//visibility:public"],
+)
